src/NServiceBus.Msmq.AcceptanceTests/When_worker_processes_a_message.cs

```diff
@@ -0,0 +1,73 @@
+﻿namespace NServiceBus.AcceptanceTests.Routing
+{
+    using System.Threading.Tasks;
+    using AcceptanceTesting;
+    using AcceptanceTesting.Customization;
+    using NServiceBus.Routing.Legacy;
+    using NUnit.Framework;
+
+    public class When_worker_processes_a_message : NServiceBusAcceptanceTest
+    {
+        static string ReceiverEndpoint => Conventions.EndpointNamingConvention(typeof(Receiver));
+
+        [Test]
+        public async Task Should_also_send_a_ready_message()
+        {
+            var context = await Scenario.Define<DistributorContext>()
+                .WithEndpoint<Receiver>()
+                .WithEndpoint<Sender>(b => b.When(c => c.WorkerSessionId != null, (s, c) =>
+                {
+                    var sendOptions = new SendOptions();
+                    sendOptions.SetHeader("NServiceBus.Distributor.WorkerSessionId", c.WorkerSessionId);
+                    return s.Send(new MyRequest(), sendOptions);
+                }))
+                .Done(c => c.ReceivedReadyMessage)
+                .Run();
+
+            Assert.IsTrue(context.ReceivedReadyMessage);
+        }
+
+        public class Sender : EndpointConfigurationBuilder
+        {
+            public Sender()
+            {
+                EndpointSetup<DefaultServer>(c =>
+                {
+                    var routing = c.UseTransport<MsmqTransport>().Routing();
+                    routing.RouteToEndpoint(typeof(MyRequest), ReceiverEndpoint);
+                });
+            }
+        }
+
+        public class Receiver : EndpointConfigurationBuilder
+        {
+            public Receiver()
+            {
+                EndpointSetup<DefaultServer>(c =>
+                {
+                    c.EnlistWithLegacyMSMQDistributor("Distributor", ReceiverEndpoint + ".Distributor", 10);
+                });
+            }
+
+            public class Detector : ReadyMessageDetector
```

Comment:
> Nope because it is out of scanning policy for ATs

that shouldn't matter if you use `EnableFeature`, as the scanning is only relevant for automatically bound classes like handlers or messages if I remember correctly.